src/Reflex/PubSub.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE ExplicitForAll        #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE RecursiveDo           #-}
+{-# LANGUAGE TypeApplications      #-}
+
+module Reflex.PubSub where
+
+import Prelude                hiding (lookup)
+
+import Control.Monad.Reader
+import Data.Constraint.Extras
+import Data.Dependent.Map
+import Data.Dependent.Sum
+import Data.Functor.Identity
+
+import Reflex
+
+type PubSubT t tag m = EventWriterT t [DSum tag Identity] (ReaderT (Event t (DMap tag Identity)) m)
```

Comment:
Suggest making this a `newtype`, so that you can define lifting instances for all the other useful classes, and people can use their own `EventWriter` or whatever else while pubsubbing.

That probably means a `Reflex.PubSub.Base` and `Reflex.PubSub.Class` pair of modules, for consistency with the rest of the library.

Reply:
Should this perhaps be a `DList (DSum tag Identity)`? I'm thinking list is probably fine, as to have it go wrong, you'd need to have a lot of `payloadE` events firing in the same frame.